### PR TITLE
Support force option for fetch_multi

### LINKED ISSUE #41535: ActiveSupport::Cache#fetch_multi does not respect the force option
### Steps to reproduce

Create a brand new rails app, `rails dev:cache` to enable caching in dev mode, then open up a console via `rails console` and paste the following:

```ruby
Rails.cache
Rails.cache.fetch_multi("foo", "bar") { |key| "original value for #{key}" }
Rails.cache.read("foo") # => "original value for foo"
Rails.cache.fetch_multi("foo", "bar", force: true) { |key| "forced value for #{key}" }
Rails.cache.read("foo") # => "original value for foo"
```

Note that this behavior is the same across all cache store implementations - we originally witnessed this in our production Rails app that uses the Redis cache store. It appears the force option was just never implemented for [fetch_multi](https://github.com/rails/rails/blob/274f8543e6e941ceba1d099e8f2081ea8fb8ca29/activesupport/lib/active_support/cache.rb#L453-L473).

### Expected behavior

The forced values should be set, and the second cache read should return `forced value for foo`

### Actual behavior

The forced values are never set, and the original values are returned:

```
=> ActiveSupport::Cache::MemoryStore entries=2, size=530, options={:compress=>false}>
=> {"foo"=>"original value for foo", "bar"=>"original value for bar"}
=> "original value for foo"
=> {"foo"=>"original value for foo", "bar"=>"original value for bar"}
=> "original value for foo"
```

### System configuration
**Rails version**:  6.1.3, but also seen in our production app running 5.2

**Ruby version**: 2.6.5

### PR DESCRIPTION
**Pull request by rsanheim**
### Summary

Fixes https://github.com/rails/rails/issues/41535 by adding support for `force` option for `fetch_multi`.

### Other Information

I'm not sure how this should impact the instrumentation, if at all -- some guidance there is appreciated.

I haven't been able to run the tests yet, as I don't have all the required dependencies installed locally for the full rails bundle. Hopefully a PR against rails can kick off the rails/rails CI?

### CODE REVIEW

**Comment by intrip** [comment]
could you please fix the rubocop failure?

**Comment by rsanheim** [comment]
@intrip Fixed!

**Comment by rsanheim** [comment]
Bump

File: activesupport/lib/active_support/cache.rb [hunk 1]
```diff
@@ -464,19 +464,26 @@ def fetch_multi(*names)
         options = names.extract_options!
         options = merged_options(options)
 
-        instrument :read_multi, names, options do |payload|
-          reads   = read_multi_entries(names, **options)
-          writes  = {}
-          ordered = names.index_with do |name|
-            reads.fetch(name) { writes[name] = yield(name) }
-          end
+        if options[:force]
```
Comment (ghiculescu) [inline activesupport/lib/active_support/cache.rb:467]: Can you document this in the above comments (same as for `#fetch`)

Comment (ghiculescu) [reply activesupport/lib/active_support/cache.rb:467]: Then once that's done could you please squash your commits and add a changelog entry.

**Comment by rafaelfranca** [comment]
Missing documentation as pointed. @rsanheim can you add the ducumentation?

**Pull request closed without merging**